### PR TITLE
Create custom role APIs

### LINKED ISSUE #2493: Add support to create / update / delete custom roles for organisations
API can be found here - https://docs.github.com/en/rest/orgs/custom-roles#create-a-custom-role

I plan on working on this feature

### PR DESCRIPTION
**Pull request by mohaldu**
resolves #2493

### CODE REVIEW

**Comment by gmlewis** [comment]
Hi @mohaldu - please don't use force-push in this repo... we always squash-and-merge anyway, so it is not necessary. See CONTRIBUTING.md for more details.

File: github/orgs_custom_roles.go [hunk 1]
```diff
@@ -44,3 +47,74 @@ func (s *OrganizationsService) ListCustomRepoRoles(ctx context.Context, org stri
...
+
+	return resultingRole, resp, err
+}
+
+// CreateCustomRepoRole creates a custom repository role in this organization
+// In order to update custom repository roles in an organization, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/en/rest/orgs/custom-roles#update-a-custom-role
+func (s *OrganizationsService) UpdateCustomRepoRole(ctx context.Context, org string, roleID string, opts *CreateOrUpdateCustomRoleOptions) (*CustomRepoRoles, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/custom_roles/%v", org, roleId)
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:85]: This one is not going to compile. Please run unit tests locally before pushing. See CONTRIBUTING.md for more information.

```suggestion
	u := fmt.Sprintf("orgs/%v/custom_roles/%v", org, roleID)
```

File: github/orgs_custom_roles.go [hunk 2]
```diff
@@ -44,3 +47,74 @@ func (s *OrganizationsService) ListCustomRepoRoles(ctx context.Context, org stri
...
+	resultingRole := new(CustomRepoRoles)
+	resp, err := s.client.Do(ctx, req, resultingRole)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return resultingRole, resp, err
+}
+
+// DeleteCustomRepoRole deletes an existing custom repository role in this organization
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:101]:
```suggestion
// DeleteCustomRepoRole deletes an existing custom repository role in this organization.
```

File: github/orgs_custom_roles.go [hunk 3]
```diff
@@ -44,3 +47,74 @@ func (s *OrganizationsService) ListCustomRepoRoles(ctx context.Context, org stri
...
+// In order to delete custom repository roles in an organization, the authenticated user must be an organization owner.
+//
+// GitHub API docs: https://docs.github.com/en/rest/orgs/custom-roles#delete-a-custom-role
+func (s *OrganizationsService) DeleteCustomRepoRole(ctx context.Context, org string, roleID string) (*Response, error) {
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:105]:
```suggestion
func (s *OrganizationsService) DeleteCustomRepoRole(ctx context.Context, org, roleID string) (*Response, error) {
```

File: github/orgs_custom_roles.go [hunk 4]
```diff
@@ -44,3 +47,74 @@ func (s *OrganizationsService) ListCustomRepoRoles(ctx context.Context, org stri
...
+		return nil, err
+	}
+
+	resultingRole := new(CustomRepoRoles)
+	resp, err := s.client.Do(ctx, req, resultingRole)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, err
```
Comment (gmlewis) [inline github/orgs_custom_roles.go:119]:
```suggestion
	return resp, nil
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @mohaldu!
Just a few changes, please, then we should be ready for a second review before merging.

**Comment by mohaldu** [comment]
Hi @gmlewis  thanks for the review, all changes made

**Comment by gmlewis** [comment]
@mohaldu - please run `go generate ./...` at the top of the repo then push the resulting changes.
See CONTRIBUTING.md for more details.

**Review by gmlewis — APPROVED** [review]
Thank you, @mohaldu !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**